src/main/java/Duke.java

```diff
@@ -1,10 +1,280 @@
+                        }
+                    }
+
+                    saveData(dest, newData, false);
+
+                    System.out.println("Noted. I've removed this task:\n" + removed.toString() +
+                            "\nNow you have " + list.size() + " tasks in the list.");
+                } else {
+                    // Adds item to list
+                    String type = input_split[0];
+                    String name;
+                    Task task;
+                    try {
+                        if (type.equals("todo")) {
+
+                            if (input_split.length < 2) {
+                                // Throw new error message that todo has no desc
+                                throw (new DukeException("\u2639 OOPS!!! The description of a todo cannot be empty."));
+                            } else {
+                                name = input.split("todo ")[1];
+                                task = new ToDo(name);
+                            }
+
+                        } else if (type.equals("deadline")) {
+                            name = input.split(" /by ")[0].split("deadline ")[1];
+                            String by = input.split(" /by ")[1];
+                            task = new Deadline(name, by);
+
+                        } else if (type.equals("event")) {
+                            name = input.split(" /at ")[0].split("event ")[1];
+                            String at = input.split(" /at ")[1];
+                            task = new Event(name, at);
+                        } else {
+                            //Throw error message for invalid input
+                            throw (new DukeException("\u2639 OOPS!!! I'm sorry, but I don't know what that means :-("));
+                        }
+                        //Add task
+                        list.add(task);
+
+                        //Append parsedData
+                        String toAppend = task.getParsedData() + System.lineSeparator();
+                        saveData(dest, toAppend, true);
+
+                        System.out.println("Go it. I've added this task:\n" + list.get(list.size() - 1).toString());
+                        System.out.println("Now you have " + list.size() + " tasks in the list.");
+
+                    } catch (DukeException e) {
+                        System.out.println(e.getMessage());
+                    }
+                }
+            }
+        } catch(IOException e) {
+            System.out.println(e.getMessage());
+        }
+
+        // Exit
+        System.out.println("Bye. Hope to see you again soon!");
+
+        sc.close();
+    }
+
+    public static void saveData(String dest, String text, Boolean appendMode) throws IOException{
+        BufferedWriter bw = new BufferedWriter(new FileWriter(dest, appendMode));
+        bw.write(text);
+        bw.close();
     }
 }
+
+
+
+class Task {
```

Comment:
Perhaps we can move these classes out of the Duke class for better organisation.